hotness/consumers.py

```diff
@@ -123,16 +105,21 @@ def __init__(self, hub):
 
         # First, initialize fedmsg and bugzilla in this thread's context.
         hostname = socket.gethostname().split('.', 1)[0]
-        fedmsg.init(name='hotness.%s' % hostname)
+        if not getattr(getattr(fedmsg, '__local', None), '__context', None):
+            fedmsg.init(name='hotness.%s' % hostname)
         fedmsg.meta.make_processors(**self.hub.config)
 
         self.bugzilla = hotness.bz.Bugzilla(
             consumer=self, config=self.config['hotness.bugzilla'])
         self.buildsys = hotness.buildsys.Koji(
             consumer=self, config=self.config['hotness.koji'])
 
-        default = 'https://admin.fedoraproject.org/pkgdb/api'
-        self.pkgdb_url = self.config.get('hotness.pkgdb_url', default)
+        default = 'https://pagure.io/releng/fedora-scm-requests'
```

Comment:
This is a pattern I've observed throughout Fedora infra code which I really don't like. 

There are a couple problems with this approach. The first problem is that often a configuration value is used in more than just one place, and can result in two (or more!) different defaults in different places in the code. The second problem is that if one wants to know what all the default values for various configurations are (or just know what configuration options are used), you have to comb the code to look for this sort of stuff.

Since you're just refactoring code that already did this, I'm okay with this being accepted as-is, but we need to file an issue about adding a configuration module that has all the valid keys and default values for them. It is also a little weird that you repeatedly overwrite ``default``. Why not just pass the string literal to ``get``?

Reply:
Agreed.  Filed as #186.